src/js/grapl-cdk/lib/service.ts

```diff
@@ -0,0 +1,193 @@
+            deadLetterQueue: { queue: dead_letter_queue, maxReceiveCount: 10 },
+            visibilityTimeout: cdk.Duration.seconds(360)
+        });
+
+        this.queue = new sqs.Queue(scope, queue_name, {
+            deadLetterQueue: { queue: this.retry_queue, maxReceiveCount: 5 },
+            visibilityTimeout: cdk.Duration.seconds(180)
+        });
+    }
+}
+
+export interface ServiceProps {
+    environment?: any,
+    vpc?: ec2.IVpc,
+    reads_from?: s3.IBucket,
+    writes_to?: s3.IBucket,
+    subscribes_to?: sns.ITopic,
+    retry_code_name?: string,
+    opt?: any
+}
+
+export class Service extends cdk.Construct {
+    readonly event_handler: lambda.Function;
+    readonly event_retry_handler: lambda.Function;
+    readonly queues: Queues
+
+    constructor(
+        scope: cdk.Construct,
+        name: string,
+        props: ServiceProps
+    ) {
+        super(scope, name + "-Service");
+
+        const environment = props.environment;
+        let retry_code_name = props.retry_code_name;
+        const opt = props.opt;
+
+        const runtime = (opt && opt.runtime) ?
+            opt.runtime : 
+            {
+                name: "provided",
+                supportsInlineCode: true
+            };
+        const handler = (runtime === lambda.Runtime.PYTHON_3_7) ?
+            `${name}.lambda_handler` :
+            name;
+
+        const queues = new Queues(this, name + '-queue');
+
+        if (environment) {
+            environment.QUEUE_URL = queues.queue.queueUrl;
+            environment.RUST_BACKTRACE = "1";
+        }
+
+        const event_handler = new lambda.Function(
+            this, name,
+            {
+                runtime: runtime,
+                handler: handler,
+                code: lambda.Code.asset(`./zips/${name}.zip`),
+                vpc: props.vpc,
+                environment: {
+                    IS_RETRY: "False",
+                    ...environment
+                },
+                timeout: cdk.Duration.seconds(180),
+                memorySize: 256,
+            });
+
+        if (!retry_code_name) {
+            retry_code_name = name
+        }
+
+        if (environment) {
+            environment.QUEUE_URL = queues.retry_queue.queueUrl;
+        }
+
+        let event_retry_handler = new lambda.Function(
+            this, name + '-retry-handler',
+            {
+                runtime: runtime,
+                handler: handler,
+                code: lambda.Code.asset(`./zips/${retry_code_name}.zip`),
+                vpc: props.vpc,
+                environment: {
+                    IS_RETRY: "True",
+                    ...environment
+                },
+                timeout: cdk.Duration.seconds(360),
+                memorySize: 512,
+            });
+
+        event_handler.addEventSource(new SqsEventSource(queues.queue, { batchSize: 1 }));
+        event_retry_handler.addEventSource(new SqsEventSource(queues.retry_queue, { batchSize: 1 }));
```

Comment:
We can make this 10 now. This was to work around a bug, and that bug has been fixed for ages.